developer/coding-guide.md

```diff
@@ -0,0 +1,65 @@
+# Coding style guide
+
+A guide to working on code at Reaction, in bite-size steps!
+
+## Git: Issues, branches, commits
+
+Git version control is how we work together as a team. Follow these guidelines to make releases and code reviews smoother.
+
+1. Find a GitHub issue for what you're working on.
+
+2. Branch name: Create a branch name, with the ID number of the GitHub issue, in the following style:
+`fix-###-yourname-slug`
+
+```
+feat-123-spencern-add-email
```

Comment:
Could we put names first so they sort together?

Reply:
The thought was to put the type first so you have all features / fixes / docs / etc together. 
Additionally, by doing that and then putting the issue no. if you somehow end up with multiple people working on the same issue, the branches will be listed next to each other.

We did discuss names first as that's what we've been doing and the general consensus between Aaron, Machiko and myself was that having branches of a type together would ultimately be more valuable